### PR TITLE
Add Ecto's `insert_or_update`

### PR DESCRIPTION
**Pull request by justanotherdot**
This adds the pure and impure variants of `insert_or_update` from Ecto (see [here](https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert_or_update/2) and [here](https://hexdocs.pm/ecto/Ecto.Repo.html#c:insert_or_update!/2)).

I have added a helper for upserting genre's into the test case but I did not see any tests for the other Ecto.Repo functions (perhaps this is worthy of a TODO?) so I've omitted any tests in this PR.

### CODE REVIEW

File: test/dal_test_case.exs [hunk 1]
```diff
@@ -122,4 +122,22 @@ defmodule DALTest.Case do
 
     %{genres: genres}
   end
+
+  def upsert_genre(_) do
```
Comment (coderdan) [inline test/dal_test_case.exs]: Is this used anywhere?

Comment (justanotherdot) [reply test/dal_test_case.exs]: No, I wasn't sure if accompanying tests should take place with this.

**Comment by justanotherdot** [comment]
> Can you please remove your vim swap files?

Sure thing. I thought I had nuked those.

**Comment by justanotherdot** [comment]
Looks like this PR is nuking the two swap files that live on master, actually.

Would you like me to remove the `upsert` helper since there are no additional tests adding it?

**Pull request merged**